.github/workflows/scripts/tag-rc.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+#
+# Add an empty commit and delete the remote tag if the tag exists.
+# Tag and push the branch.
+#
+# Local wet run:
+#
+#   DRY_RUN=false bash local-env.sh tag-rc <tag>
+#
+set -euo pipefail
+
+TAG="$1"
+
+check_not_empty \
+    TAG \
+    \
+    DRY_RUN
+
+if git ls-remote --tags --exit-code origin "$TAG"; then
+    git push --delete origin "$TAG"
+    git commit --allow-empty --message "Empty commit to trigger CI"
+    gh_log warning "Tag '$TAG' has been deleted and an empty commit has been added to trigger CI."
+fi
+
+git tag --force --annotate "$TAG" --message "Upstream release automation"
+
+if [ "$DRY_RUN" = "false" ]; then
+    git push --follow-tags
+fi
```

Comment:
There are 2 cases that are uncovered yet, but we want to cover here:

1. `$TAG` is unknown yet, but there were no commits created in cherry-picking and no updates to the submodules.
2. Last commit on the release branch is untagged (e.g., a fix was merged to the release branch but not to `master` - this has happened once in 3.71)

To cover these both cases, we could do:

```suggestion
if git ls-remote --tags --exit-code origin "$TAG"; then
    git push --delete origin "$TAG"
    gh_log warning "Tag '$TAG' has been deleted"
fi

git commit --allow-empty --message "Empty commit to trigger CI"
gh_log warning "An empty commit has been added to trigger CI."
git tag --force --annotate "$TAG" --message "Upstream release automation"

if [ "$DRY_RUN" = "false" ]; then
    git push --follow-tags
fi
```

WDYT?

Reply:
Right, but let's make it simpler, see above.

Reply:
It feels strange to review a comment suggestion that is an answer to a comment suggestion, because I cannot comment inline :) 

The important thing is that we want to remove the tag ONLY if it is on the tip of the branch. If you run `git ls-remote --tags --exit-code origin "3.69.1-rc.4"` then it will return 0 and then you would simply delete something that is not the tip of the branch.

I would suggest to do the following:
- fetch branch from origin
- check if the last commit is tagged 
    - if not, then add empty commit and tag it
    - if yes, then is that tag the same as `$TAG`?
        - if yes, then remove the tag, add empty commit and tag it
        - if no, then only add empty commit and tag it

Reply:
Please take a look at the updated version. I simply check the HEAD for the tag before deleting it.